runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -818,10 +818,33 @@ Java_java_lang_invoke_MethodHandleNatives_resolve(JNIEnv *env, jclass clazz, job
 
 				/* Check for resolution exception */
 				if (VM_VMHelpers::exceptionPending(currentThread)) {
-					goto done;
-				}
-
-				if (NULL != method) {
+					J9Class *exceptionClass = J9OBJECT_CLAZZ(currentThread, currentThread->currentException);
+					if ((ref_kind == MH_REF_INVOKESPECIAL) && (exceptionClass == J9VMJAVALANGINCOMPATIBLECLASSCHANGEERROR(vm))) {
+						/* Special handling for default method conflict, defer the exception throw until invocation. */
+						VM_VMHelpers::clearException(currentThread);
+						/* Attempt to lookup method without checking for conflict.
+						 * If this failed with error, then throw that error.
+						 * Otherwise exception throw will be defered to the MH.invoke call.
+						 */
+						method = lookupMethod(
+							currentThread, resolvedClass, name, signature, callerClass,
+							(lookupOptions & ~J9_LOOK_HANDLE_DEFAULT_METHOD_CONFLICTS));
+
+						if (!VM_VMHelpers::exceptionPending(currentThread)) {
+							/* Set placeholder values for MemberName fields. */
+							vmindex = (jlong)(UDATA)1;
```

Comment:
Why `1`?  Could we store the `method` in `vmindex` and use it to create a better exception message?

Reply:
the vmindex / vmtarget fields are lost after we finished the `invokeBasic/linkTo` INLs as memberName is droped from the stack, `throwDefaultConflict` is used as target to avoid extra checking during the INL call for perfermance.

The 1 is stored as memberName will be treated as unresolved if vmindex field is NULL, also using 1 makes it eaiser to differentiate with other resolved methods as this is a special case without a JNI ref.

Reply:
Can you name the constant. Like `J9VM_RESOLVED_VMINDEX_FOR_DEFAULT_THROW`